### ISSUE TITLE
AutoincrementTest failing

### ISSUE THREAD

**Issue by aguibert** [hibernate/hibernate-reactive#125]
The `Autoincrement` test is failing, which means it has something to do with `@GeneratedValue(strategy = GenerationType.IDENTITY)` specifically for PostgreSQL (the other PG tests are working fine, and the MySQLAutoincrementTest is also working).

@gavinking do you have time to look into this one? 

Here is the full stack trace (although it's not too helpful)
```
Gradle Test Executor 20 > org.hibernate.rx.AutoincrementTest > testBasicTypes FAILED                                                                                                                                                             java.util.concurrent.CompletionException: io.vertx.pgclient.PgException: relation "basic" does not exist                                                                                                                                         
at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)                                                                                                                                                        
at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)                                                                                                                                                      
at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:607)                                                                                                                                                               
at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)                                                                                                                                                       
at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)                                                                                                                                                           
at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)                                                                                                                                                 
at org.hibernate.rx.impl.PoolConnection.lambda$toCompletionStage$5(PoolConnection.java:106)                                                                                                                                                  
at org.hibernate.rx.impl.PoolConnection.lambda$null$1(PoolConnection.java:80)                                                                                                                                                                
at io.vertx.sqlclient.impl.SqlClientBase$PreparedQueryImpl.lambda$execute$0(SqlClientBase.java:113)                                                                                                                                          
at io.vertx.sqlclient.impl.PoolBase$1.lambda$onSuccess$0(PoolBase.java:102)                                                                                                                                                                  
at io.vertx.sqlclient.impl.SocketConnectionBase.handleMessage(SocketConnectionBase.java:188)                                                                                                                                                 
at io.vertx.sqlclient.impl.SocketConnectionBase.lambda$init$0(SocketConnectionBase.java:78)                                                                                                                                                  
at io.vertx.core.net.impl.NetSocketImpl.lambda$new$2(NetSocketImpl.java:101)                                                                                                                                                                 
at io.vertx.core.streams.impl.InboundBuffer.handleEvent(InboundBuffer.java:237)                                                                                                                                                              
at io.vertx.core.streams.impl.InboundBuffer.write(InboundBuffer.java:127)                                                                                                                                                                    
at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:357)                                                                                                                                                                
at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:366)                                                                                                                                                                          
at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:43)                                                                                                                                                                     
at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:229)                                                                                                                                                                        
at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:173)                                                                                                                                                                    
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)                                                                                                                                  
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)                                                                                                                                  
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)                                                                                                                                    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)                                                                                                      at io.vertx.pgclient.impl.codec.PgEncoder.lambda$write$0(PgEncoder.java:78)                                                                                                                                                                  at io.vertx.pgclient.impl.codec.PgCommandCodec.handleReadyForQuery(PgCommandCodec.java:136)                                                                                                                                                  at io.vertx.pgclient.impl.codec.PrepareStatementCommandCodec.handleReadyForQuery(PrepareStatementCommandCodec.java:76)                                                                                                                       at io.vertx.pgclient.impl.codec.PgDecoder.decodeReadyForQuery(PgDecoder.java:227)                                                                                                                                                            at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:86)                                                                                                                                                                     at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)                                                                                                                                          
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)                                                                                                                                  
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)                                                                                                                                  
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)                                                                                                                                    
at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)                                                                                                                                         
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)                                                                                                                                  
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)                                                                                                                                  
at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)                                                                                                                                                  
at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)                                                                                                                                           
at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)                                                                                                                                                               
at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)                                                                                                                                                     
at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)                                                                                                                                                              
at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)                                                                                                                                                                              
at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)                                                                                                                                              
at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)                                                                                                                                                                 
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)                                                                                                                                                     
at java.lang.Thread.run(Thread.java:748)                                                                                                                                                                                                                                                                                                                                                                                                                                                  
Caused by:                                                                                                                                                                                                                                   io.vertx.pgclient.PgException: relation "basic" does not exist                                                                                                                                                                                   
at io.vertx.pgclient.impl.codec.ErrorResponse.toException(ErrorResponse.java:29)                                                                                                                                                             
at io.vertx.pgclient.impl.codec.PrepareStatementCommandCodec.handleErrorResponse(PrepareStatementCommandCodec.java:70)                                                                                                                       
at io.vertx.pgclient.impl.codec.PgDecoder.decodeError(PgDecoder.java:233)                                                                                                                                                                    
at io.vertx.pgclient.impl.codec.PgDecoder.decodeMessage(PgDecoder.java:122)                                                                                                                                                                  
at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:102)                                                                                                                                                                    
... 17 more                                                                           
```

**Comment by gavinking** [comment]
Well the MySQL tests are failing only when running in Docker. When I run them against my local MySQL database they all run successfully.

**Comment by gavinking** [comment]
It seems to be that the testcontainers thing wants to have the JDBC driver, and I removed that dependency. No idea what it needs the driver for.

```
Gradle Test Executor 2 > org.hibernate.rx.MySQLAutoincrementTest > testBasicTypes FAILED
    org.testcontainers.containers.ContainerLaunchException: Container startup failed
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:322)
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:302)
        at org.hibernate.rx.containers.MySQLDatabase.getJdbcUrl(MySQLDatabase.java:24)
        at org.hibernate.rx.MySQLAutoincrementTest.constructConfiguration(MySQLAutoincrementTest.java:47)

        Caused by:
        org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
            at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
            at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:315)
            ... 3 more

            Caused by:
            org.testcontainers.containers.ContainerLaunchException: Could not create/start container
                at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:485)
                at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:317)
                at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
                ... 4 more

                Caused by:
                org.testcontainers.containers.JdbcDatabaseContainer$NoDriverFoundException: Could not get Driver
                    at org.testcontainers.containers.JdbcDatabaseContainer.getJdbcDriverInstance(JdbcDatabaseContainer.java:169)
                    at org.testcontainers.containers.JdbcDatabaseContainer.createConnection(JdbcDatabaseContainer.java:191)
                    at org.testcontainers.containers.JdbcDatabaseContainer.waitUntilContainerStarted(JdbcDatabaseContainer.java:129)
                    at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:428)
                    ... 6 more

                    Caused by:
                    java.lang.ClassNotFoundException: com.mysql.jdbc.Driver
                        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
                        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
                        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
                        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
                        at java.lang.Class.forName0(Native Method)
                        at java.lang.Class.forName(Class.java:264)
                        at org.testcontainers.containers.JdbcDatabaseContainer.getJdbcDriverInstance(JdbcDatabaseContainer.java:167)
                        ... 9 more
```

**Comment by DavideD** [comment]
Should we just add back the dependencies?

**Comment by DavideD** [comment]
@aguibert You mentioned that you can write a custom check. I don't like too much the idea of having  custom stuff for this but how much work would that be?

**Comment by aguibert** [comment]
@gavinking @DavideD yes, the MySQLContainer tries to establish a JDBC connection to determine container readiness. I fixed this in one of the commits on my DB2 PR, which I need to rebase on all of the Rx->Reactive rename changes

**Comment by aguibert** [comment]
@DavideD the custom check is pretty simple, you can see it here: https://github.com/hibernate/hibernate-reactive/pull/117/commits/c39950d60f467a8a9de9c0c7cd99314d7e511f5d#diff-4006a0aa525560f05c2d5491e9cbd07c
Basically just trying to get a Vertx connection in a loop and then attempting the same test query SQL string that is used for the JDBC check

**Comment by DavideD** [comment]
`pretty simple` :)

So, this is only a problem for the MySQL db?

**Comment by aguibert** [comment]
yes, the other containers (DB2 and Postgresql) use container log message checks to determine container readiness

**Comment by DavideD** [comment]
ok

**Comment by aguibert** [comment]
@gavinking I've merged in the DB2 support PR now, which also contains the fix for the issue you are seeing. So now you can reproduce this issue on `master` by doing:
`./gradlew :hibernate-reactive-core:test -Pdocker --tests AutoincrementTest`

**Comment by gavinking** [comment]
Excellent, thanks!

**Comment by aguibert** [comment]
Fixed by #141
